packages/zoe/src/contractFacet/contractFacet.js

```diff
@@ -191,6 +214,69 @@ export function buildRootObject(powers, _params, testJigSetter = undefined) {
       return { zcfSeat, userSeat: userSeatPromiseKit.promise };
     };
 
+    /** @type {MakeInvitation} */
+    const makeInvitation = (
+      offerHandler = () => {},
+      description,
+      customProperties = {},
+    ) => {
+      assert.typeof(
+        description,
+        'string',
+        details`invitations must have a description string: ${description}`,
+      );
+
+      const invitationHandle = makeHandle('InvitationHandle');
+      invitationHandleToHandler.init(invitationHandle, offerHandler);
+      /** @type {Promise<Payment>} */
+      const invitation = E(zoeInstanceAdmin).makeInvitation(
+        invitationHandle,
+        description,
+        customProperties,
+      );
+      return invitation;
+    };
+
+    /** @type {MakeInvitationWithSeat} */
+    const makeInvitationWithSeat = async (
+      offerHandler,
+      description,
+      customProperties,
+      exit,
+    ) => {
+      const invitation = makeInvitation(
+        offerHandler,
+        description,
+        customProperties,
+      );
+      const invitationAmount = await invitationIssuer.getAmountOf(invitation);
+      // AWAIT ///
+
+      const {
+        updater,
+        zoeSeatAdminPromiseKit,
+        userSeatPromiseKit,
+        seatHandle,
+        zcfSeat,
+      } = initializeEmptySeatData(exit);
+      const invitationHandle = invitationAmount.value[0].handle;
+      invitationHandleToZcfSeatKit.init(invitationHandle, {
+        zcfSeatAdmin: zcfSeatToZCFSeatAdmin.get(zcfSeat),
+        zcfSeat,
+      });
+
+      await E(zoeInstanceAdmin)
+        .makeLaterEscrowSeat(seatHandle, invitationHandle)
+        .then(({ zoeSeatAdmin, notifier: zoeNotifier, userSeat }) => {
+          updateFromNotifier(updater, zoeNotifier);
```

Comment:
The `await` on your line 268 is awaiting the result of the `updateFromNotifier`, which I suspect is not what you want. The promise returned by `updateFromNotifier` doesn't settle until either `finish` or `fail` is called on the updater for this notifier. I suspect you want to await one step earlier, in which case you could write it as

```suggestion
      const { zoeSeatAdmin, notifier: zoeNotifier, userSeat } = 
        await E(zoeInstanceAdmin).makeLaterEscrowSeat(seatHandle, invitationHandle);
      // AWAIT ///

      updateFromNotifier(updater, zoeNotifier);
```
and then proceed to return the `invitation,userSeat` kit in the same turn in which you *call* `updateFromNotifier`. I don't think anything in your logic needs to care about the promise returned by `updateFromNotifier`.

Reply:
Agreed that this could be clearer, but the curlies around the original `.then` callback means that the call to `updateFromNotifier` is not awaited.

Reply:
@erights comment appeared elsewhere for some reason. In order to reply at the right place, I'll copy it here.  erights replied: 
> Oops. You are correct. This is an interesting misreading hazard I was not really aware of. I'll keep it in mind. Thanks.

I read this as acknowledging that the current code is fine. The `await` ensures that the promises are resolved before we return.